### ISSUE TITLE
Proposal - refactor into 3 smaller samples

### ISSUE THREAD

**Issue by paulyuk** [Azure-Samples/Azure-Functions-OpenAI-Demo#24]
The proposal is to split this very good sample into 3 smaller fit for purpose samples.  The current sample is great but it is too complex, too hard to set up locally, and too hard to find.  By doing this proposal, we will have easy to search samples that each do one job well. 

Proposed smaller samples:
1) ChatGPT Chatbot - using assistants
2) RAG - using embeddings output and semantic search
3) Function calling - using Assistant and Assistant Skills trigger

Note, I observe we are already down this path.  e.g. 
* Chat samples exist for [python](https://github.com/Azure-Samples/function-python-ai-openai-chatgpt) and [javascript](https://github.com/Azure-Samples/function-javascript-ai-openai-chatgpt)
* Function calling exists at least for [C#](https://github.com/eamonoreilly/azure-functions-assistants-dotnet8) in eamon's profile

So we'd just need to fill out matrix a bit more with RAG and the other languages.

For languages I recommend:
P1 - Python, Javascript/Node, C#/Dotnet
P2 - TypeScript, PowerShell

@eamonoreilly @BigMorty @MadhuraBharadwaj-MSFT please weigh in

**Comment by paulyuk** [comment]
I also wonder if the SWA frontend UI is worth the complexity (all the extra files and dev ops tasks/scripts to deploy).  I could see this demo still existing for full end to end, whereas the small samples are lightweight.  would love opinions.

**Comment by BigMorty** [comment]
Definitely agree in splitting this sample into three samples, that will make them much easier to grasp for folks new to our AI extension.
Regarding the SWA frontend, agree we should just use http files like we do in the other samples, assuming we can do the file upload fairly easy still.

**Comment by MadhuraBharadwaj-MSFT** [comment]
Yes, agreed on this proposal. It makes sense to do a logical split and have them in smaller, easier to consume scenarios. I'm currently working on the RAG + AI Search sample for JavaScript.
